crates/signature-mpc/src/twopc_mpc_protocols/encrypt_user_share.rs

```diff
@@ -70,6 +72,19 @@ pub fn generate_keypair() -> Result<(Vec<u8>, Vec<u8>)> {
     Ok((encryption_key, decryption_key))
 }
 
+/// Hash the `bytes` using Keccak256, and generate a keypair for the
+/// Paillier encryption scheme deterministically from that hash.
+pub fn generate_keypair_from_bytes(bytes: &[u8]) -> Result<(Vec<u8>, Vec<u8>)> {
+    let mut hasher = sha3::Keccak256::new();
+    hasher.update(bytes);
+    let hashed = hasher.finalize();
+    let mut rng = StdRng::from_seed(<[u8; 32]>::from(hashed));
```

Comment:
Use https://rust-random.github.io/rand/rand_chacha/struct.ChaCha20Core.html

Reply:
When trying to use it, I get 
```
error[E0277]: the trait bound `ChaCha20Core: CryptoRngCore` is not satisfied
  --> crates/signature-mpc/src/twopc_mpc_protocols/encrypt_user_share.rs:84:68
   |
84 |     let (encryption_key, decryption_key) = DecryptionKey::generate(&mut rng)?;
   |                                            ----------------------- ^^^^^^^^ the trait `RngCore` is not implemented for `ChaCha20Core`, which is required by `ChaCha20Core: CryptoRngCore`
   |                                            |
   |                                            required by a bound introduced by this call
   |
   = note: required for `ChaCha20Core` to implement `CryptoRngCore`
note: required by a bound in `DecryptionKey::generate`
  --> /Users/itaylevy/.cargo/git/checkouts/tiresias-007e763c08f3070e/24930c9/src/decryption_key.rs:29:24
   |
28 |     pub fn generate(
   |            -------- required by a bound in this associated function
29 |         rng: &mut impl CryptoRngCore,
   |                        ^^^^^^^^^^^^^ required by this bound in `DecryptionKey::generate`
```

Should I edit the `tiresias` `generate` function to support this random generator?